tests/test_batch.py

```diff
@@ -217,5 +219,335 @@ def given_configuration_file_version_different_as_river_file_version_when_runnin
             assert str(cm.value) == 'Version number of configuration file (2.0) must match version number of rivers file (1.0)'
         finally:
             os.chdir(cwd)
+            
+class Test_batch_save_configuration_file():      
+    def given_configuration_file_when_save_configuration_file_then_file_is_saved_with_expected_data(self, tmp_path):
+        expected_lines = [
+        "[General]\n",
+        "  riverkm     = RiverKM\n",
+        "  figuredir   = FigureDir\n",
+        "  outputdir   = OutputDir\n",
+        "\n",
+        "[SomeSection]\n",
+        "  reference   = reference\n",
+        "  withmeasure = with_measure\n",
+        ]
+        
+        file_path = tmp_path / "test_file.cfg"
+        config = self.sample_config(tmp_path)
+        
+        dfastmi.batch.save_configuration_file(file_path, config)
+        
+        assert os.path.exists(file_path)
+        with open(file_path, 'r') as file:
+            file_lines = file.readlines()
+            
+        assert len(file_lines) == len(expected_lines)
+        assert file_lines == expected_lines
+        
+    def sample_config(self, tmp_path):
+        # Create a sample configuration
+        config = ConfigParser()
+        config["General"] = {
+            "RiverKM": tmp_path / "RiverKM",
+            "FigureDir": tmp_path / "FigureDir",
+            "OutputDir": tmp_path / "OutputDir"
+        }
+        config["SomeSection"] = {
+            "Reference": tmp_path / "reference",
+            "WithMeasure": tmp_path / "with_measure"
+        }
+        return config
+    
+class Test_batch_countq():       
+    @pytest.mark.parametrize("vector_data, expected_non_empty_discharges_count", [
+        ([0.123, None, 0.456, 0.789, None], 3),
+        ([0.123, 0.123, 0.456, 0.789, 0.123], 5),
+        ([None, None, None, None, None], 0),
+    ])    
+    def given_vector_with_discharges_when_countq_then_return_amount_of_non_empty_discharges(self, vector_data: Vector, expected_non_empty_discharges_count: int):
+        assert dfastmi.batch.countQ(vector_data) == expected_non_empty_discharges_count
+ 
+class Test_batch_write_report():   
+    @pytest.mark.parametrize("slength", [
+        0.202122,
+        1.202122
+    ])       
+    def given_input_data_with_varying_slength_when_write_report_then_expect_messages_written_in_report(self, slength :float):
+        report = StringIO()
+        reach = "reach"
+        q_location = "location"
+        q_threshold = 0.123
+        q_bankfull = 0.456
+        q_stagnant = 0.789
+        tstag = 0.101112
```

Comment:
why 6 numbers behind comma? double is double right :-) 
it is test data. may be using 0.1 for first, 0.2 for second, etc sufficient? just for readability

Reply:
I wasnt sure what would happen with each value during testing, so I wanted to give every variable a distinct value.
Such I started with 123, 456 and ended with the last on 101112, but yeah, they can be changed to move easily read numbers behind the comma, will update.

Reply:
done